authenticate/handlers.go

```diff
@@ -603,3 +608,67 @@ func (a *Authenticate) saveSessionToDataBroker(
 
 	return nil
 }
+
+// FrontchannelLogout signs the user out
+func (a *Authenticate) FrontchannelLogout(w http.ResponseWriter, r *http.Request) error {
+	ctx, span := trace.StartSpan(r.Context(), "authenticate.FrontchannelLogout")
+	defer span.End()
+
+	w.Header().Set("Content-Type", "text/plain; charset=utf-8")
+	w.Header().Set("Cache-Control", "no-cache, no-store")
+	w.Header().Set("Pragma", "no-cache")
+
+	if err := a.processFrontchannelLogoutWithContext(ctx, w, r); err != nil {
+		log.FromRequest(r).Debug().Err(err).Msg("failed processing frontchannel logout")
+
+		w.WriteHeader(http.StatusBadRequest)
+		fmt.Fprintf(w, "%s", "mising required parameters")
+
+		return nil
+	}
+
+	w.WriteHeader(http.StatusOK)
+	fmt.Fprintf(w, "%s", "OK")
+
+	return nil
+}
+
+func (a *Authenticate) processFrontchannelLogoutWithContext(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
```

Comment:
This function does a very similar (identical?) process as the logout function. Cand we abstract and resuse that code as part of a subfunction?

https://github.com/pflipp/pomerium/blob/d0cfab7020816098a73426d3234a22cdb182634a/authenticate/handlers.go#L247-L263